aten/src/ATen/Utils.h

```diff
@@ -73,9 +73,6 @@ static inline TensorImpl* checked_tensor_unwrap(const Tensor& expr, const char *
     AT_ERROR("Expected object of scalar type ", scalar_type, " but got scalar type ", expr.scalar_type(),
              " for argument #", pos, " '", name, "'");
   }
-  if (expr.is_variable()) {  // TODO: change this to check `.requires_grad()` and `GradMode::is_enabled()` when Variable and Tensor are merged
-    AT_ERROR("Expected Tensor (not Variable) for argument #", pos, " '", name, "'");
-  }
```

Comment:
We don't need this check (and the one below in `checked_tensor_list_unwrap()`) anymore, because its original purpose was to make sure we are not calling `expr.unsafeGetTensorImpl()` on a Variable. Now that since both Variable and Tensor share the same TensorImpl class, using `expr.unsafeGetTensorImpl()` is always correct and we don't need the check anymore.